### PR TITLE
We need SLE-Module-Containers on the Pod Proxy VM

### PR DESCRIPTION
**Pull request by szachovy**
## What does this PR change?

Adds `Containers Module 15 SP4 x86_64` channels to Pod Proxy VM and Retail Branch Server as mandatory ones during SCC product synchronization, to rely not only on the SLES15 SP4 product `podman` packages.

## GUI diff

`Containers Module 15 SP4 x86_64` child channel of `SUSE Manager Proxy 4.3 x86_64` and `SUSE Manager Retail Branch Server 4.3 x86_64` is now selected in Setup Wizard products.

Screenshot from 2024-02-21 12-37-32

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23240
Port(s): https://github.com/SUSE/spacewalk/pull/23757

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature [hunk 1]
```diff
@@ -711,25 +711,6 @@ Feature: Synchronize products in the products page of the Setup Wizard
     When I use spacewalk-common-channel to add channel "debian-12-pool-amd64-uyuni debian-12-amd64-main-updates-uyuni debian-12-amd64-main-security-uyuni debian-12-amd64-uyuni-client-devel" with arch "amd64-deb"
     And I wait until all synchronized channels for "debian-12" have finished
 
-@susemanager
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature:714]: yes

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature [hunk 2]
```diff
@@ -742,25 +723,6 @@ Feature: Synchronize products in the products page of the Setup Wizard
     When I use spacewalk-common-channel to add all "leap15.5" channels with arch "x86_64"
     And I wait until all synchronized channels for "uyuni-proxy" have finished
 
-@susemanager
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature:745]: and yes

**Review by Bischoff — CHANGES_REQUESTED** [review]

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (szachovy) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature]: Ok, fragment restored.

File: testsuite/features/reposync/srv_sync_products.feature [hunk 3]
```diff
@@ -118,25 +118,33 @@ Feature: Synchronize products in the products page of the Setup Wizard
...
+    Then I should see the "Containers Module 15 SP4 x86_64" selected
     When I click the Add Product button
     And I wait until I see "Selected channels/products were scheduled successfully for syncing." text
     And I wait until I see "SUSE Manager Proxy 4.3 x86_64" product has been added
     And I wait until all synchronized channels for "suma-proxy-43" have finished
 
 @proxy
 @susemanager
-  Scenario: Add SUSE Manager Retail Branch Server
+  Scenario: Add SUSE Manager Retail Branch Server 4.3
```
Comment (Bischoff) [inline testsuite/features/reposync/srv_sync_products.feature:132]: oh, thanks! :smile_cat: 

but this scenario (and the previous one) will create a problem with SUSE Manager 5.0 :disappointed: .

should we change all 4.3 to 5.0 (only in uyuni branch) ? Ping @nodeg .

File: testsuite/features/reposync/srv_sync_products.feature [hunk 4]
```diff
@@ -118,25 +118,33 @@ Feature: Synchronize products in the products page of the Setup Wizard
...
     Then I should see the "SUSE Manager Retail Branch Server 4.3 x86_64" selected
+    When I open the sub-list of the product "SUSE Manager Retail Branch Server 4.3 x86_64"
+    And I open the sub-list of the product "Basesystem Module 15 SP4 x86_64"
+    And I select "Containers Module 15 SP4 x86_64" as a product
+    Then I should see the "Containers Module 15 SP4 x86_64" selected
     When I click the Add Product button
     And I wait until I see "Selected channels/products were scheduled successfully for syncing." text
     And I wait until I see "SUSE Manager Retail Branch Server 4.3 x86_64" product has been added
-    And  I wait until all synchronized channels for "suma-retail-branch-server-43" have finished
+    And I wait until all synchronized channels for "suma-retail-branch-server-43" have finished
```
Comment (Bischoff) [inline testsuite/features/reposync/srv_sync_products.feature:147]: nice catch

File: testsuite/features/support/constants.rb [hunk 5]
```diff
@@ -951,13 +951,23 @@
         sle-product-suse-manager-proxy-4.3-updates-x86_64
         sle-module-suse-manager-proxy-4.3-pool-x86_64
         sle-module-suse-manager-proxy-4.3-updates-x86_64
+        sle-module-basesystem15-sp4-pool-x86_64-proxy-4.3
```
Comment (Bischoff) [inline testsuite/features/support/constants.rb:954]: Thank you.

**Review by Bischoff — APPROVED** [review]
approving, but see a question about a problem unveiled here

(assuming we fix it, will be resolved in another PR anyway)

File: testsuite/features/reposync/srv_sync_products.feature (hunk 3 above)
Comment (nodeg) [reply testsuite/features/reposync/srv_sync_products.feature:132]: I have no strong opinion about that. We can diverge with this between SUMA and Uyuni.

Comment (Bischoff) [reply testsuite/features/reposync/srv_sync_products.feature:132]: It was rather beteen suma 4.3 and suma 5.0. But I guess I already answered myself, we'll just switch when 5.0 is out.

Comment (szachovy) [reply testsuite/features/reposync/srv_sync_products.feature:132]: I also have no strong opinion about that.

**Pull request merged**